pygama/dsp/_processors/get_multi_local_extrema.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+from numba import guvectorize, njit
+from pygama.dsp.errors import DSPFatal
+
+
+
+@guvectorize(["void(float32[:], float32, float32[:], float32[:], float32[:], float32[:], float32[:], int32[:])",
+              "void(float64[:], float64, float64[:], float64[:], float64[:],float64[:], float64[:], int32[:])",
+              "void(int32[:], int32, int32[:], int32[:], int32[:], int32[:], int32[:], int32[:])",
+              "void(int64[:], int64, int64[:], int64[:], int64[:], int64[:], int64[:], int32[:])"],
+             "(n),(),(m) -> (m),(m),(),(),()", nopython=True, cache=True)
+def get_multi_local_extrema(w_in, delta_in, t_arbitrary_in, vt_max_out, vt_min_out, n_max_out, n_min_out, flag_out):
+    """
+    Get lists of indices of the local maxima and minima of data
+    The "local" extrema are those maxima / minima that have heights / depths of
+    at least delta.
+    Converted from MATLAB script at: http://billauer.co.il/peakdet.html
+    Parameters
+    ----------
+    w_in : array-like
+        The array of data within which extrema will be found
+    delta_in : scalar
+        The absolute level by which data must vary (in one direction) about an
+        extremum in order for it to be tagged
+    t_arbitrary_in : array-like
+        An array of fixed length that tells numba to expect to return a list of amplitudes of the same length
+    Returns
+    -------
+    vt_max_out, vt_min_out : array-like, array-like
+        Arrays of fixed length (padded with nans) that hold the indices of
+        the identified local maxima and minima
+    n_max_out, n_min_out: scalar, scalar 
+        The number of maxima and minima found in a waveform
+    flag_out: scalar
+        Returns 0 if there is only one maximum and it is a simple waveform, 
+        Returns 1 if there are multiple peaks in a waveform
+        
+    """
+
+    # prepare output 
+    #padded with nans or zeros depending on what we want
+    vt_max_out[:]= np.nan
+    vt_min_out[:] = np.nan
+
+    
+    n_max_out[0] = 0
+    n_min_out[0] = 0
+    flag_out[0] = 0
```

Comment:
I would set all these output parameters to nan and return 0 only if the processor succeeds